### PR TITLE
Add exportJSON to ArtificialNode

### PR DESCRIPTION
**Pull request by thegreatercurve**
Fix browser warnings for missing method on ArtificialNode.

### CODE REVIEW

**Review by potatowagon — APPROVED** [review]
thanks lgtm

File: packages/lexical/src/nodes/ArtificialNode.ts [hunk 1]
```diff
@@ -20,4 +20,8 @@ export class ArtificialNode__DO_NOT_USE extends ElementNode {
     const dom = document.createElement('div');
     return dom;
   }
+
+  exportJSON() {
+    return super.exportJSON();
```
Comment (zurfyx) [inline packages/lexical/src/nodes/ArtificialNode.ts:25]: Let's make it throw an invariant instead please. This Node should not exist and should never land on any database.

**Review by zurfyx — CHANGES_REQUESTED** [review]

File: packages/lexical/src/nodes/ArtificialNode.ts (hunk 1 above)
Comment (etrepum) [reply packages/lexical/src/nodes/ArtificialNode.ts:25]: I have a commit in #6048 that just ignores the warnings for these nodes, which is another approach that would preserve expected runtime errors without sacrificing DX https://github.com/facebook/lexical/pull/6048/commits/8cd0664de7613d7a9cf6b0ed86cac1d56f2fd48e

Comment (potatowagon) [reply packages/lexical/src/nodes/ArtificialNode.ts:25]: im not sure if changing to invariant would continue to raise errors. the browser warnings was reported to break the serialisation of the editor state on KP's usecase (T189214869), hence the goal of the PR to silence the warnings.

Comment (potatowagon) [reply packages/lexical/src/nodes/ArtificialNode.ts:25]: > I have a commit in #6048 that just ignores the warnings for these nodes, which is another approach that would preserve expected runtime errors without sacrificing DX [8cd0664](https://github.com/facebook/lexical/commit/8cd0664de7613d7a9cf6b0ed86cac1d56f2fd48e)

thanks for this, lgtm!

**Comment by thegreatercurve** [comment]
Closing in favour of #6048

**Pull request closed without merging**